core/src/main/java/org/apache/calcite/rel/core/JoinRelType.java

```diff
@@ -94,6 +150,37 @@ public JoinRelType cancelNullsOnRight() {
       return this;
     }
   }
+
+  /** Transform this JoinRelType to CorrelateJoinType. **/
+  public CorrelateJoinType toLinq4j() {
+    switch (this) {
+    case INNER:
+      return CorrelateJoinType.INNER;
+    case LEFT:
+      return CorrelateJoinType.LEFT;
+    case SEMI:
+      return CorrelateJoinType.SEMI;
+    case ANTI:
+      return CorrelateJoinType.ANTI;
+    }
+    throw new IllegalStateException(
+        "Unable to convert " + this + " to CorrelateJoinType");
+  }
+
+  public boolean projectsRight() {
+    switch (this) {
+    case INNER:
+    case LEFT:
+    case RIGHT:
+    case FULL:
+      return true;
+    case SEMI:
+    case ANTI:
+      return false;
+    }
+    throw new IllegalStateException(
```

Comment:
It seems that this `throw` statement was copy-pasted and does not belong here.

Reply:
There is no conversion here, the exception looks weird.

Reply:
I copied it from `SemiJoinType#toLinq4j`, why do you think it is weird ? I think it is necessary cause `Correlate` could not have `RIGHT` as join type.

Reply:
But here we are in method `projectsRight` not in toLinq4j am I missing something?

Reply:
@zabetak Oops, i mistaken it, but if i remove the `throw` java compiler will complains error. We have 2 choice:
1. throw exception (cause we may extend the join type in future)
2. return a default value (i don't think we actually can have the default here)

So, 1 is more acceptable. What do you think ?

Reply:
Let's go for 1.

Reply:
Thanks @zabetak, let's just keep is as now.